src/coreclr/jit/importer.cpp

```diff
@@ -18912,40 +18912,47 @@ void Compiler::impMakeDiscretionaryInlineObservations(InlineInfo* pInlineInfo, I
         }
     }
 
-    if ((info.compMethodInfo->args.callConv & CORINFO_CALLCONV_GENERIC) ||
-        (info.compClassAttr & CORINFO_FLG_SHAREDINST))
-    {
-        inlineResult->Note(InlineObservation::CALLEE_GENERIC);
-    }
-    else if ((rootCompiler->info.compMethodInfo->args.callConv & CORINFO_CALLCONV_GENERIC) ||
-             (rootCompiler->info.compClassAttr & CORINFO_FLG_SHAREDINST))
+    bool callsiteIsGeneric = (rootCompiler->info.compMethodInfo->args.callConv & CORINFO_CALLCONV_GENERIC) ||
+                             (rootCompiler->info.compClassAttr & CORINFO_FLG_SHAREDINST);
```

Comment:
I don't believe `CORINFO_CALLCONV_GENERIC` is the right check here.

See eg https://github.com/dotnet/coreclr/pull/11780

Reply:
Changed to `sigInst.methInstCount != 0` (previous one also worked on my tests)